src/hb_http_signature.erl

```diff
@@ -381,3 +559,342 @@ random_an_binary(Length) ->
 	RandomIndexes = [rand:uniform(ListLength) || _ <- lists:seq(1, Length)],
 	RandomChars = [lists:nth(Index, Characters) || Index <- RandomIndexes],
 	list_to_binary(RandomChars).
+
+%%% @doc Recursively trim space characters from the beginning of the binary
+trim_ws(<<$\s, Bin/bits>>) -> trim_ws(Bin);
+%%% @doc No space characters at the beginning so now trim them from the end
+%%% recrusively
+trim_ws(Bin) -> trim_ws_end(Bin, byte_size(Bin) - 1).
+
+trim_ws_end(_, -1) ->
+	<<>>;
+trim_ws_end(Value, N) ->
+	case binary:at(Value, N) of
+		$\s ->
+			trim_ws_end(Value, N - 1);
+		% No more space characters matches on the end
+		% So extract the bytes up to N, and this is our trimmed value
+		_ ->
+			S = N + 1,
+			<<Trimmed:S/binary, _/bits>> = Value,
+			Trimmed
+	end.
+
+%%%
+%%% TESTS
+%%%
+
+trim_ws_test() ->
+	<<"hello world">> = trim_ws(<<"      hello world      ">>),
```

Comment:
I would suggest using the ?assertEqual, as the usual way of verifying with erlang unit tests. For example the output from match verification is:
```
in function hb_http_signature:trim_ws_test/0 (/Users/oleg/repos/arweave/HyperBEAM/src/hb_http_signature.erl, line 588)
in call from eunit_test:'-mf_wrapper/2-fun-0-'/2 (eunit_test.erl, line 274)
in call from eunit_test:run_testfun/1 (eunit_test.erl, line 72)
in call from eunit_proc:run_test/1 (eunit_proc.erl, line 544)
in call from eunit_proc:with_timeout/3 (eunit_proc.erl, line 369)
in call from eunit_proc:handle_test/2 (eunit_proc.erl, line 527)
in call from eunit_proc:tests_inorder/3 (eunit_proc.erl, line 469)
in call from eunit_proc:with_timeout/3 (eunit_proc.erl, line 359)
**error:{badmatch,<<"hello world     \t">>}
  output:<<"">>
```
(aka, some badmatch on line 588 :))

With ?assertEqual you have:
```
hb_http_signature: trim_ws_test...*failed*
in function hb_http_signature:trim_ws_test/0 (/Users/oleg/repos/arweave/HyperBEAM/src/hb_http_signature.erl, line 588)
in call from eunit_test:'-mf_wrapper/2-fun-0-'/2 (eunit_test.erl, line 274)
in call from eunit_test:run_testfun/1 (eunit_test.erl, line 72)
in call from eunit_proc:run_test/1 (eunit_proc.erl, line 544)
in call from eunit_proc:with_timeout/3 (eunit_proc.erl, line 369)
in call from eunit_proc:handle_test/2 (eunit_proc.erl, line 527)
in call from eunit_proc:tests_inorder/3 (eunit_proc.erl, line 469)
in call from eunit_proc:with_timeout/3 (eunit_proc.erl, line 359)
**error:{assertEqual,[{module,hb_http_signature},
              {line,588},
              {expression,"trim_ws ( << \"      hello world    \" >> )"},
              {expected,<<"hello world  !">>},
              {value,<<"hello world">>}]}
  output:<<"">>
```

so: We tried to call trim_ws ( << \"      hello world    \" >> ), got <<"hello world">> expected: <<"hello world  !">>

Reply:
I've moved all the tests to use the `assertEqual` macro